### PR TITLE
Add vis_label

### PR DESCRIPTION
**Pull request by Hakuyume**
merge after #195

### CODE REVIEW

**Comment by Hakuyume** [comment]
```
from chainercv.datasets import VOCSemanticSegmentationDataset
from chainercv.visualizations import vis_image
from chainercv.visualizations import vis_label
dataset = VOCSemanticSegmentationDataset()
img, label = dataset[60]
ax = vis_image(img)
vis_label(label, alpha=0.75, ax=ax)
```
seg_sample

**Comment by Hakuyume** [comment]
I'm trying to show a legend using `label_names`.

File: chainercv/visualizations/vis_label.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+def vis_label(label, alpha=1, ax=None):
+    """Visualize label of semantic segmentation.
```
Comment (yuyu2172) [inline chainercv/visualizations/vis_label.py]: a label for semantic segmantation

File: chainercv/visualizations/vis_label.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        >>> img, label = dataset[60]
+        >>> ax = vis_image(img)
+        >>> vis_label(label, alpha=0.75, ax=ax)
+        >>> plot.show()
+
+    Args:
+        label (~numpy.ndarray): An integer array of shape
+            :math:`(height, width)`.
+            The values correspond to id for label names stored in
+            :obj:`label_names`.
```
Comment (yuyu2172) [inline chainercv/visualizations/vis_label.py]: label_names are not given.
Perhaps, simply put it as "The values corresponds to classes assigned to pixels"?

File: chainercv/visualizations/vis_label.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        alpha (float): The value which determines transparency of the figure.
+            The range of this value is :math:`[0, 1]`. If this
+            value is :obj:`0`, the figure will be completely transparent.
+            The default value is :obj:`1`. This option is useful for
+            overlaying labels on the source images.
```
Comment (yuyu2172) [inline chainercv/visualizations/vis_label.py]: the source image

**Comment by yuyu2172** [comment]
Depending on datasets, colors assigned to class ids differ.
I think we should change `vis_label` to accept colors.

**Comment by yuyu2172** [comment]
Since labels are in range [-1, n_fg_class], how about accepting two arguments `label_colors` and `ignore_color`. `label_colors` is a (n_fg_class + 1, 3) color array and `ignore_color is a (3,) color array (RGB).
In most cases, `ignore_color` is (0, 0, 0)  (i.e. black).

**Comment by yuyu2172** [comment]
Or, we can just pass a dict with integer keys and color array as values.

**Comment by yuyu2172** [comment]
Here is a visualization code used in SegNet.
https://github.com/alexgkendall/SegNet-Tutorial/blob/master/Scripts/test_segmentation_camvid.py#L48-L60

**Comment by Hakuyume** [comment]
> Depending on datasets, colors assigned to class ids differ.
I think we should change vis_label to accept colors.

Yes, I am planning to do so.

**Comment by Hakuyume** [comment]
```
import numpy as np
from chainercv.datasets import VOCSemanticSegmentationDataset
from chainercv.datasets import voc_semantic_segmentation_label_names
from chainercv.visualizations import vis_image
from chainercv.visualizations import vis_label
colors = np.random.randint(0, 256, size=(21, 3))
dataset = VOCSemanticSegmentationDataset()
img, label = dataset[60]
ax = vis_image(img)
_, legend_handles = vis_label(label, label_names=voc_semantic_segmentation_label_names, colors=colors, alpha=0.9, ax=ax)
ax.legend(handles=legend_handles, bbox_to_anchor=(1, 1), loc=2)
```
vizseg

**Comment by Hakuyume** [comment]
> Depending on datasets, colors assigned to class ids differ.

Do you know the set of colors for PASCAL VOC? I think it is better to add `voc_semantic_segmentation_label_colors`

**Comment by Hakuyume** [comment]
```
import numpy as np
from chainercv.datasets import VOCSemanticSegmentationDataset
from chainercv.datasets import voc_semantic_segmentation_label_names
from chainercv.visualizations import vis_image
from chainercv.visualizations import vis_label
label_colors = np.random.randint(0, 256, size=(21, 3))
dataset = VOCSemanticSegmentationDataset()
img, label = dataset[60]
ax = vis_image(img)
_, legend_handles = vis_label(label, label_names=voc_semantic_segmentation_label_names, label_colors=label_colors, alpha=0.9, ax=ax)
ax.legend(handles=legend_handles, bbox_to_anchor=(1, 1), loc=2, ncol=2)
```
vizseg

**Comment by Hakuyume** [comment]
I found the definition of the official color map in the orginal MATLAB code.
I will add this as `voc_semantic_segmentation_label_colors`.

**Comment by yuyu2172** [comment]
> I found the definition of the official color map in the orginal MATLAB code. I will add this as voc_semantic_segmentation_label_colors.

Wow. Great!

**Comment by Hakuyume** [comment]
I added `voc_semantic_segmentation_label_colors`.

**Comment by Hakuyume** [comment]
```
from chainercv.datasets import VOCSemanticSegmentationDataset
from chainercv.datasets import voc_semantic_segmentation_label_colors
from chainercv.datasets import voc_semantic_segmentation_label_names
from chainercv.visualizations import vis_image
from chainercv.visualizations import vis_label
dataset = VOCSemanticSegmentationDataset()
img, label = dataset[60]
ax = vis_image(img)
_, legend_handles = vis_label(
    label,
    label_names=voc_semantic_segmentation_label_names,
    label_colors=voc_semantic_segmentation_label_colors,
    ignore_label_color=(255, 255, 255),
    alpha=0.9, ax=ax)
ax.legend(handles=legend_handles, bbox_to_anchor=(1, 1), loc=2, ncol=2)
```
vizseg

**Comment by yuyu2172** [comment]
Is (255, 255, 255) used in the PASCAL MATLAB code?
I am just curious.

**Comment by Hakuyume** [comment]
> Is (255, 255, 255) used in the PASCAL MATLAB code?
I am just curious.

No. In the MATLAB code, they use `(224, 224, 192)` for the ignored label and `(0, 0, 0)` for `background`.

**Comment by yuyu2172** [comment]
I see.

File: chainercv/visualizations/vis_label.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        ...     import voc_semantic_segmentation_label_colors
+        >>> from chainercv.datasets \
+        ...     import voc_semantic_segmentation_label_names
+        >>> from chainercv.visualizations import vis_image
+        >>> from chainercv.visualizations import vis_label
+        >>> import matplotlib.pyplot as plot
+        >>> dataset = VOCSemanticSegmentationDataset()
+        >>> img, label = dataset[60]
+        >>> ax = vis_image(img)
+        >>> _, legned_handles = vis_label(
```
Comment (yuyu2172) [inline chainercv/visualizations/vis_label.py]: legend_handles

**Comment by yuyu2172** [comment]
I think default colors map the same class id in different images to different colors.

```python
from chainercv.visualizations import vis_label
import numpy as np
import matplotlib.pyplot as plot



img0 = np.zeros((32, 32), dtype=np.int32)

img0[16:] = 2
img0[:16] = 1

img1 = np.zeros((32, 32), dtype=np.int32)

img1[16:] = 0
img1[:16] = 1


fig = plot.figure()
ax1 = fig.add_subplot(1, 2, 1)
vis_label(img0, ax=ax1)
ax2 = fig.add_subplot(1, 2, 2)
vis_label(img1, ax=ax2)
plot.show()
```

**Comment by yuyu2172** [comment]
I think default color is quite tricky to make it work.
We can work on this in the future by making the argument optional.

**Comment by Hakuyume** [comment]
I changed the default color map.
```
from chainercv.visualizations import vis_label
import numpy as np
import matplotlib.pyplot as plot

img0 = np.zeros((32, 32), dtype=np.int32)
img0[16:] = 2
img0[:16] = 1

img1 = np.zeros((32, 32), dtype=np.int32)
img1[16:] = 0
img1[:16] = 1

fig = plot.figure()
ax1 = fig.add_subplot(1, 2, 1)
vis_label(img0, ax=ax1)
ax2 = fig.add_subplot(1, 2, 2)
vis_label(img1, ax=ax2)
plot.show()
```
vizseg

**Comment by yuyu2172** [comment]
Can you add `(224, 224, 192)` as `voc_semantic_segmentation_ignore_label_color`?

**Comment by Hakuyume** [comment]
> Can you add (224, 224, 192) as voc_semantic_segmentation_ignore_label_color?

OK. I added.

**Comment by yuyu2172** [comment]
LGTM

**Pull request merged**